internal/xpkg/snapshot/validators.go

```diff
@@ -157,6 +161,15 @@ func validatorsFromV1Beta1XRD(x *xpextv1beta1.CompositeResourceDefinition, acc m
 }
 
 func validatorsFromV1XRD(x *xpextv1.CompositeResourceDefinition, acc map[schema.GroupVersionKind]*validator.ObjectValidator) error {
+	errs := validateOpenAPIV3Schema(x)
+	if errs != nil {
```

Comment:
```suggestion
	if len(errs) != 0 {
```

Current implementation panics on zero-length slice.